test/integration/policy/policy_test.go

```diff
@@ -0,0 +1,94 @@
+package policy
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-network-policy-agent/test/framework/manifest"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	network "k8s.io/api/networking/v1"
+)
+
+var _ = Describe("Network Policy Test Cases", func() {
+	Context("A network policy is applied against a pod with '.' in its pod name", func() {
+
+		var clientPod *v1.Pod
+		var clientNetworkPolicy *network.NetworkPolicy
+		var podName = "this.pod.has.dots.in.its.name"
+
+		BeforeEach(func() {
+			By("Creating a network policy that denies all egress traffic", func() {
+				clientNetworkPolicy = manifest.NewNetworkPolicyBuilder().
+					Namespace(namespace).
+					Name("deny-all-egress").
+					PodSelector("app", podName).
+					SetPolicyType(false, true).
+					Build()
+
+				err := fw.NetworkPolicyManager.CreateNetworkPolicy(ctx, clientNetworkPolicy)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			By("Creating a pod which tries to reach external network", func() {
+				agnhostContainer := manifest.NewAgnHostContainerBuilder().
+					ImageRepository(fw.Options.TestImageRegistry).
+					Args([]string{"while true; do wget https://www.google.com --spider -T 1; if [ $? == 0 ]; then echo \"Success\"; else echo \"Fail\"; fi; sleep 1s; done"}).
+					Build()
+
+				clientPod = manifest.NewDefaultPodBuilder().
+					Container(agnhostContainer).
+					Namespace(namespace).
+					AddLabel("app", podName).
+					Name(podName).
+					Build()
+
+				_, err := fw.PodManager.CreateAndWaitTillPodIsRunning(ctx, clientPod, 2*time.Minute)
+				Expect(err).ToNot(HaveOccurred())
+			})
+		})
+
+		It("should be successfully applied and removed from the pod", func() {
+			By("Verifying that the pod is unable to make an egress connection", func() {
+				time.Sleep(1 * time.Minute)
+				logs, err := fw.PodManager.PodLogs(namespace, podName)
+				Expect(err).ToNot(HaveOccurred())
+				err = processLogsForEndState(logs, "Fail")
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			By("Removing the network policy", func() {
+				fw.NetworkPolicyManager.DeleteNetworkPolicy(ctx, clientNetworkPolicy)
+			})
+
+			By("Verifying that the pod is once again able to make an egress connection", func() {
+				time.Sleep(1 * time.Minute)
+				logs, err := fw.PodManager.PodLogs(namespace, podName)
+				Expect(err).ToNot(HaveOccurred())
+				err = processLogsForEndState(logs, "Success")
+				Expect(err).ToNot(HaveOccurred())
+			})
+		})
+
+		AfterEach(func() {
+			fw.PodManager.DeleteAndWaitTillPodIsDeleted(ctx, clientPod)
+		})
+
+	})
+
+})
+
+func processLogsForEndState(podlogs string, state string) error {
+
+	logSlice := strings.Split(strings.TrimSuffix(podlogs, "\n"), "\n")
+	endState := logSlice[len(logSlice)-1]
+
+	if endState != state {
```

Comment:
This will not work always as the last string will not always be "Success" / "Fail".

Reply:
For my first test case where I apply a `deny-all-egress` policy and spin up the pod, I saw logs like...
```
Success
Success
Success
Fail
Fail
...
Fail
```
so the logs would start out as `Success`, and then the `deny-all-egress` network policy is applied to the pod, and the rest of the logs would become a long tail of `Fail`. 

I agree that if there is some perturbation of the system then the `Fail` could switch to `Success` and the test would seem to fail. 
One way to ensure that the test is genuinely detecting success or failure could be to use [MustPassRepeatedly()](https://onsi.github.io/ginkgo/#repeating-spec-runs-and-managing-flaky-specs), which will run the spec up to N times until they fail.

Reply:
Actually, I have an idea for how to make this work. For this test case I'll validate it with a state machine that expects the logs to begin with "Success" and to end with "Fail". I'll implement similar logic for the second policy removal spec too